freqtrade/commands/arguments.py

```diff
@@ -65,7 +65,7 @@
 
 ARGS_CONVERT_TRADES = ["pairs", "timeframes", "exchange", "dataformat_ohlcv", "dataformat_trades"]
 
-ARGS_LIST_DATA = ["exchange", "dataformat_ohlcv", "pairs"]
+ARGS_LIST_DATA = ["exchange", "dataformat_ohlcv", "pairs", "trading_mode"]
```

Comment:
Will this limit getting the mark/index price candles to what trading mode your using? I feel like they could be useful for spot, just as a reference

Reply:
well here - surely not - as this line is only relevant for `list-data`.

the way i think about this currently is as follows:
you run `freqtrade download-data --exchange <whatever> --pairs XRP/USDT BTC/USDT` to get spot data (this is how it is also in develop).

For futures, you'll run `freqtrade download-data --exchange <whatever> --pairs XRP/USDT BTC/USDT --trading-mode futures` - which will download 2 things, the "futures" candles (stored in `user_data/data/binance/futures/XRP_USDT-futures.json` - and mark candles (or index, depending on the exchange - stored as `user_data/data/binance/futures/XRP_USDT-mark.json`).

For the moment, i think freqtrade will take the decision on which candles we'll download (so if backtesting needs 1h mark candles, we'll download exactly this).

From a usability perspective, i think this "automatism" will help, as the alternative would be to have users run 2 "download-data" commands, one for "futures" - and one for "mark"  (using the correct timeframe for the exchange used - the one we require for backtesting) - which doesn't really make sense, as the candles are mandatory for backtesting to work (and the only reason to download data is really to run backtesting ...).

I guess we COULD have this as argument to download-data too (so you can only download mark / index / whatever candles) - but it's not a necessity for now, and we'll have to think about usability here very carefully, as you need to use "futures" mode to even get mark candles (otherwise the exchange object is initialized with the wrong default type).